packages/shell-chrome/src/state.js

```diff
@@ -1,185 +1,185 @@
 import { flattenData, convertInputDataToType } from "./utils";
 
 export default class State {
-  constructor() {
-    this.components = {};
-    this.allDataAttributes = {};
-    this.renderedComponentId = null;
-  }
-
-  renderComponentsFromBackend(components) {
-    components.forEach((component, index) => {
-      component.index = index;
-      component.isOpened = this.renderedComponentId == component.id;
-
-      component.flattenedData = flattenData(component.data);
-
-      component.flattenedData.forEach((d) => {
-        if (
-          (this.allDataAttributes[component.id] &&
-            this.allDataAttributes[component.id][d.id] &&
-            this.allDataAttributes[component.id][d.id].isOpened) ||
-          (d.directParentId.length &&
-            this.allDataAttributes[component.id][d.directParentId] &&
-            this.allDataAttributes[component.id][d.directParentId].isArrowDown)
-        ) {
-          d.isOpened = true;
-        }
+    constructor() {
+        this.components = {};
+        this.allDataAttributes = {};
+        this.renderedComponentId = null;
+    }
 
-        if (
-          this.allDataAttributes[component.id] &&
-          this.allDataAttributes[component.id][d.id] &&
-          this.allDataAttributes[component.id][d.id].hasArrow
-        ) {
-          d.isArrowDown = this.allDataAttributes[component.id][
-            d.id
-          ].isArrowDown;
-        }
+    renderComponentsFromBackend(components) {
+        components.forEach((component, index) => {
+            component.index = index;
+            component.isOpened = this.renderedComponentId == component.id;
+
+            component.flattenedData = flattenData(component.data);
+
+            component.flattenedData.forEach((d) => {
```

Comment:
This is a pretty gnarly forEach loop, is there anything else we can do to make it tidier?

Reply:
I wouldn't touch it until we've got some sort of tests in place